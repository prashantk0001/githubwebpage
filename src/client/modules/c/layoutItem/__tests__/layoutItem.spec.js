/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import cElement from 'c/layoutItem';

function createComponent(props = {}) {
    const element = createElement('c-layout-item', {
        is: cElement
    });

    Object.assign(element, props);
    document.body.appendChild(element);
    return element;
}

describe('c-layout-item', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('default', () => {
        const element = createComponent({
            size: 12
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('flexibility', () => {
        const element = createComponent({
            size: 12,
            flexibility: 'auto,grow,shrink'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('alignment-bump', () => {
        const element = createComponent({
            size: 12,
            alignmentBump: 'right'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('padding', () => {
        const element = createComponent({
            size: 12,
            padding: 'around-medium'
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
