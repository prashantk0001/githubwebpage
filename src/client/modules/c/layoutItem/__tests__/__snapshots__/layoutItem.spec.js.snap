// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-layout-item alignment-bump 1`] = `
<c-layout-item
  class="slds-size_12-of-12 slds-col_bump-right"
>
  #shadow-root(open)
    <slot />
</c-layout-item>
`;

exports[`c-layout-item default 1`] = `
<c-layout-item
  class="slds-size_12-of-12"
>
  #shadow-root(open)
    <slot />
</c-layout-item>
`;

exports[`c-layout-item flexibility 1`] = `
<c-layout-item
  class="slds-col slds-grow slds-shrink slds-size_12-of-12"
>
  #shadow-root(open)
    <slot />
</c-layout-item>
`;

exports[`c-layout-item padding 1`] = `
<c-layout-item
  class="slds-p-around_medium slds-size_12-of-12"
>
  #shadow-root(open)
    <slot />
</c-layout-item>
`;
